### DISCUSSION TITLE
Using MFRONT in Python

### DISCUSSION THREAD

**Discussion by NsmKsm** [thelfer/tfel#598] [General]
Hello, 

I am implementing an elasto-plastic problem in python and I want to use Mfront as an integration function (for behaviour). I noticed that Mfront generates the shared library ''libumatbehaviour.so'' alongside cxx files (in the src folder). 

My questions:
 - Is it possible to call the shared library in python ? (maybe using "ctypes").
 - Is there a way to use Mfront as an integration function in python ?

Thanks in advance, 
Bests,

**Comment by thelfer** [comment]
Dear @NsmKsm,
There are two ways to do this, depending on your needs.

1. For a fully favoured solver, consider using MGIS: https://github.com/thelfer/MFrontGenericInterfaceSupport (see also the [JOSS paper]()https://www.theoj.org/joss-papers/joss.02003/10.21105.joss.02003.pdf for details).
2. For a more lightweight solution, consider using the mtest module: https://thelfer.github.io/tfel/web/mtest-python.html

In both cases, you need to enable the `python` bindings which rely on the `Boost/Python` library.
Best,

**Reply by NsmKsm** [reply]
Thank you.

**Reply by thelfer** [reply]
Let me know if you need anything else
